### DISCUSSION TITLE
Is it possible to add penality to vehicle?

### DISCUSSION THREAD

**Discussion by ajinra020307** [google/or-tools#3404] [Routing (and legacy CP) questions]
I am trying to use or tools for a vehicle routing problem. I want to add penality to some vehicles , so the solver use those vehicles less. Is there any way we can achieve this in ortools?

Thanks

**Comment by Mizux** [comment]
yes you can, we call it the Vehicle Fixed cost, aka if solver use this vehicle then it must pay the VehicleFixedCost otherwise it's 0 so solver has incentive to avoid to use it...

ref: https://github.com/google/or-tools/blob/d493335e968c73952d96d3edbc1ee9229fce033d/ortools/constraint_solver/routing.h#L1038-L1046
